python/ray/serve/metrics.py

```diff
@@ -0,0 +1,34 @@
+from ray.util import metrics
+from typing import Tuple, Optional, Dict
+
+REPLICA_TAG = "N/A"
+DEPLOYMENT = "N/A"
+
+
+class Counter(metrics.Counter):
+    def __init__(
+        self, name: str, description: str = "", tag_keys: Optional[Tuple[str]] = None
+    ):
+        # Make sure customer tags are not using the serve default tags.
+        assert "deployment" not in tag_keys
+        assert "replica" not in tag_keys
```

Comment:
this is a user-facing validation error so it should not use `assert`. Use `raise ValueError("helpful message")` instead.